pyanalyze/implementation.py

```diff
@@ -416,6 +416,14 @@ def _dict_setitem_impl(ctx: CallContext) -> ImplReturn:
     return ImplReturn(KnownValue(None))
 
 
+def _list_getitem_impl(ctx: CallContext) -> ImplReturn:
+    self_value = ctx.vars['self']
+    index = ctx.vars['index']
+    if isinstance(index, TypedValue):
+        if index.typ is slice:
+            return TypedValue(self_value.get_type())
```

Comment:
`.get_type()` shouldn't be used as it's imprecise. In particular, here it's going to drop the type parameter.

Reply:
What should I use instead, `.typ` ?

Reply:
In most cases I think returning the `self_value` itself is actually correct: a slice over a `List[int]` is another `List[int]`.

It gets a bit more complicated with `SequenceIncompleteValue`, where we know exactly which values are in the list or tuple. In that case, if we know precisely what the slice is we can return a more precise `SequenceIncompleteValue`. A simpler approach would be to fall back to the `GenericValue` that it inherits from and return that.

Reply:
> In most cases I think returning the self_value itself is actually correct

Good point. But it's not correct for KnownValue

I couldn't find a way to get the values of the slice.